var/spack/repos/builtin/packages/py-pennylane-lightning-kokkos/package.py

```diff
@@ -71,11 +72,12 @@ class PyPennylaneLightningKokkos(CMakePackage, PythonExtension, CudaPackage, ROC
     depends_on("py-pybind11", type="link")
     depends_on("py-pip", type="build")
     depends_on("py-wheel", type="build")
-    depends_on("py-pennylane@0.30:", type=("build", "run"), when="@0.31:")
     depends_on("py-pennylane@0.28:", type=("build", "run"), when="@:0.30.0")
-    depends_on("py-pennylane-lightning@0.31:~kokkos", type=("build", "run"), when="@0.31.0:")
-    depends_on("py-pennylane-lightning@0.30:~kokkos", type=("build", "run"), when="@0.30.0:")
-    depends_on("py-pennylane-lightning@0.28:0.29~kokkos", type=("build", "run"), when="@0.29.1")
+    depends_on("py-pennylane@0.30:", type=("build", "run"), when="@0.31:")
+    depends_on("py-pennylane-lightning~kokkos", type=("build", "run"), when="@:0.31")
+    depends_on("py-pennylane-lightning@0.28:0.29", type=("build", "run"), when="@0.29.1")
+    for v in range(30, 33):
```

Comment:
For version 0.30, I see a redundant dependency definition on `py-pennylane-lightning` between lines 77 and 79-80:
* py-pennylane-lightning~kokkos
* py-pennylane-lightning

Shouldn't line 79 only range betwen 31,33? Or otherwise get rid of line 77.

Reply:
Line 77 is to add `~kokkos` to the specs when ver <= 0.31.
The lines below are to specify the Lightning versions.
They work independently, which avoids conditionally inserting `~kokkos` in the specs (won't work for 0.32).

Reply:
I am not sure I am convinced, I would appreciate an extra clarification.

Let me substitute line 80 with `v=30`, and compare with line 77:

```
# line 77
depends_on("py-pennylane-lightning~kokkos", type=("build", "run"), when="@:0.31")

# line 80, with `v = 30`
 depends_on("py-pennylane-lightning@0.30:", type=("build", "run"), when="@0.30.0:")
```

To me, both these specifications apply to versions `0.30.x`, isn't it?
So, why have two?
If specification for this version series is different from future on, why not having it a special case (i.e. line 77) and remove it from the version loop below?

Thank you,

Reply:
> Never mind, I got it now!
> 
> Thanks for your patience, please consider this one as resolved!

NP. To clarify why we have `py-pennylane-lightning~kokkos` in the first place. `py-pennylane-lightning` is pulled as a dependency of `py-pennylane`, but truly it is not really required to run `py-pennylane-lightning-kokkos`. Before 0.32, `py-pennylane-lightning` has `kokkos` & `kokkos-kernels` by default, but it is expecting a host-only backend (and officially the serial-only backend actually). I found `py-pennylane-lightning+kokkos` to make concretization much more difficult in practice (sometimes impossible, for me at least). Since `py-pennylane-lightning` is not really required, `py-pennylane-lightning-kokkos` builds the simpler version of `py-pennylane-lightning` which makes it easier to deal with all possible `kokkos` backends and options.